opal/mca/threads/qthreads/threads_qthreads_module.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include <unistd.h>
+
+#include "opal/constants.h"
+#include "opal/util/sys_limits.h"
+#include "opal/util/output.h"
+#include "opal/prefetch.h"
+#include "opal/mca/threads/threads.h"
+#include "opal/mca/threads/tsd.h"
+
+struct opal_tsd_key_value {
+    opal_tsd_key_t key;
+    opal_tsd_destructor_t destructor;
+};
+
+static pthread_t opal_main_thread;
+struct opal_tsd_key_value *opal_tsd_key_values = NULL;
+static int opal_tsd_key_values_count = 0;
+
+/*
+ * Constructor
+ */
+static void opal_thread_construct(opal_thread_t *t)
+{
+    t->t_run = 0;
+    t->t_handle = (pthread_t) -1;
+}
+
+OBJ_CLASS_INSTANCE(opal_thread_t,
+                   opal_object_t,
+                   opal_thread_construct, NULL);
+
+
+opal_thread_t *opal_thread_get_self(void)
+{
+    opal_thread_t *t = OBJ_NEW(opal_thread_t);
+    t->t_handle = pthread_self();
+    return t;
+}
+
+bool opal_thread_self_compare(opal_thread_t *t)
+{
+    return t->t_handle == pthread_self();
```

Comment:
Are the calls in this module support to be for P threads or Q threads?

Reply:
This is just a placeholder using the pthreads module for now.

Reply:
There's zero explanation of this in the code.

If you want to have a no-op component, fine, do so.  But don't call it a real name and put in a whole bunch of code that makes it look like it's doing something.  If someone didn't bother to read the code, this would have gone and and anyone who ran `ompi_info` would have thought that we supported Q threads.

Reply:
Fair point. I'll put in the qthreads testing code with some caveats in the comments (it requires a custom experimental version of qthreads with nonblocking synchronization and thread local storage).

Reply:
Please either add comments in *this* file -- or a pointer to where you have a full explanation (e.g., a README or comments in some other file -- no need to copy/have the full explanation more than once) -- of why there are `pthreads` calls in the `qthreads` component.  I says this because it's an obvious question that will occur to anyone who is reading the code and isn't deeply familiar with the qthreads code base.